internal/lifecycle/readme.go

```diff
@@ -79,17 +79,17 @@ func (cb codeBlock) toCommands(serviceName, gcrURL string) ([]*exec.Cmd, error)
 			line = line + l
 		}
 
-		line = os.ExpandEnv(line)
 		line = gcrURLRegexp.ReplaceAllString(line, gcrURL)
 		line = replaceServiceName(line, serviceName)
-		sp := strings.Split(line, " ")
 
 		var cmd *exec.Cmd
-		if sp[0] == "gcloud" {
-			a := append(util.GcloudCommonFlags, sp[1:]...)
-			cmd = exec.Command("gcloud", a...)
+		if gcloudCommandRegexp.MatchString(line) {
+			sp := strings.Split(line, " ")
+			c := "gcloud " + strings.Join(append(util.GcloudCommonFlags, sp[1:]...), " ")
```

Comment:
I think you may want to avoid strings.Join here, and instead just pass it as arguments into exec.Command and let it concat it incase anything special needs to be done on different OSs

Reply:
Thank you for your comment.

The reason we joined the string back up was because I believe `bash` expects the entire command as a single argument. Testing this on the command line, `bash -c echo $HELLO_WORLD` didn't print anything whereas `bash -c "echo $HELLO_WORLD"` printed the expected value. Similarly, testing this in Go,the output of `exec.Command("bash", "-c", "echo", "$HELLO_WORLD")` was nothing and output of `exec.Command("bash", "-c", "echo $HELLO_WORLD")` was as expected.

This is part of the reason we hadn't implemented support for other shells yet, since we're not sure which executable and flags we need to use for running a command through the shell and how that executable expects the command to be passed to it.